presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJoinQueriesWithPushDown.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.jdbc;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.presto.tpch.TpchPlugin;
+import com.google.common.collect.ImmutableList;
+import io.airlift.tpch.TpchTable;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static com.facebook.airlift.testing.Closeables.closeAllSuppress;
+import static com.facebook.presto.SystemSessionProperties.INEQUALITY_JOIN_PUSHDOWN_ENABLED;
+import static com.facebook.presto.SystemSessionProperties.INNER_JOIN_PUSHDOWN_ENABLED;
+import static com.facebook.presto.plugin.jdbc.JdbcQueryRunner.createSchema;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tests.QueryAssertions.copyTpchTables;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+
+public class TestJoinQueriesWithPushDown
```

Comment:
While this is a good integration test for the feature using H2, we can test more complex inner-join pushdown scenarios by creating a product test where we do joins between multiple Hive and JDBC (say MySQL) tables. This test also takes a long time to execute (~66s) because of H2's ability to execute large joins 

A recent example of a MySQL based product test is [this test](https://github.com/prestodb/presto/commit/2ba9e2740c0d579f1ef09fe18d77289875d86603#diff-25b4cffd4a81d84cdaad735a14b0c7f5f8588e7027be3dec482f3ea78b799a86) from the mixed-case feature

Can you explore adding a similar one ?